### PR TITLE
fix: fixed hot reload on React Native

### PR DESCRIPTION
**Pull request by icc-romeu**
This is a fix for RN Hot reload. I picked @TimoGlastra work and fixed it for iOS -it was not working- and Android.

The problem was that React Native reloads the native modules each time you reload the app and usually Metro will do that each time you save a file. Hence, the Wallet and Pool handles are lost on the native module, but they are still opened in libindy, and you got an error.

What I did is defining the wallet and the pool handles as static dictionaries, and store them in memory. They cannot be class members as IndySDK class is re-created each time RN reloads.

I fixed some deprecation warnings on iOS too.

### CODE REVIEW

**Comment by icc-romeu** [comment]
@JamesKEbert maybe this is interesting for you if you are using it on Aries Bifold

**Comment by icc-romeu** [comment]
This fixes https://github.com/AbsaOSS/rn-indy-sdk/issues/41

File: android/src/main/java/com/reactlibrary/IndySdkModule.java [hunk 1]
```diff
@@ -60,21 +62,15 @@
     private static final String TAG = "IndySdk";
     private final ReactApplicationContext reactContext;
 
-    private Map<Integer, Wallet> walletMap;
-    private Map<Integer, Pool> poolMap;
-    private Map<Integer, WalletSearch> searchMap;
-    private Map<Integer, CredentialsSearchForProofReq> credentialSearchMap;
-
-    // Java wrapper does not expose credentialSearchHandle
-    private int credentialSearchIterator = 0;
```
Comment (TimoGlastra) [inline android/src/main/java/com/reactlibrary/IndySdkModule.java]: Is there a specific reason you removed these? Seems unrelated to this change

**Review by TimoGlastra — COMMENTED** [review]
Thanks @icc-romeu, you finished what I couldn't :-)

File: android/src/main/java/com/reactlibrary/IndySdkModule.java (hunk 1 above)
Comment (icc-romeu) [reply android/src/main/java/com/reactlibrary/IndySdkModule.java]: oops! No! Good catch! Let me fix it! :)

**Comment by icc-romeu** [comment]
Fixed your comment @TimoGlastra. Ready to be merged if you want to 😄

**Comment by TimoGlastra** [comment]
Looks good to me! I don't have permission to merge however.

We need @jakubkoci for that

**Comment by JamesKEbert** [comment]
Thanks for your additions here @icc-romeu and @TimoGlastra !
